server/routes/azure/azure_routes.py

```
@@ -147,31 +145,38 @@ def azure_clusters(user_id):
+            logging.warning(f"[AZURE API] No Azure subscription found for user {user_id}")
+            return jsonify({"error": "No Azure subscription found. Please configure your Azure subscription."}), 401
+        projects = [{"projectId": subscription_id, "name": subscription_name, "enabled": True}]
+        return jsonify({"projects": projects}), 200
+    except Exception as e:
+        logging.error("Error in azure_subscriptions_get", exc_info=e)
+        return jsonify({"error": "Failed to process Azure subscriptions"}), 500
+
+
+@azure_bp.route("/api/azure-subscriptions", methods=["POST"])
+@require_permission("connectors", "write")
+def azure_subscriptions_post(user_id):
```

Ruff (0.15.11)

[warning] Unused function argument: `user_id`

(ARG001)